client/components/search/index.jsx

```diff
@@ -38,7 +38,8 @@ module.exports = React.createClass( {
 		onKeyDown: React.PropTypes.func,
 		disableAutocorrect: React.PropTypes.bool,
 		onBlur: React.PropTypes.func,
-		searching: React.PropTypes.bool
+		searching: React.PropTypes.bool,
+		forceCloseButton: React.PropTypes.bool
```

Comment:
Aha, so the case here is that "add" works as the search icon in `pinned` mode. So maybe what you need is to be able to set `isOpen` to true initially, which will render the close button. @bluefuton does that make sense? I'd rather have a `forceOpen` or similar that sets the initial state of `isOpen` since it's more semantic.